src/neuroconv/datainterfaces/behavior/medpc/medpcdatainterface.py

```diff
@@ -0,0 +1,248 @@
+from typing import Optional
+
+import numpy as np
+from pynwb.behavior import BehavioralEpochs, IntervalSeries
+from pynwb.file import NWBFile
+
+from neuroconv.basetemporalalignmentinterface import BaseTemporalAlignmentInterface
+from neuroconv.tools import get_package, nwb_helpers
+from neuroconv.utils import DeepDict, FilePathType
+
+from .medpc_helpers import read_medpc_file
+
+
+class MedPCInterface(BaseTemporalAlignmentInterface):
+    """Data Interface for MedPC output files.
+
+    The output files from MedPC are raw text files that contain behavioral data from the operant box sessions such as
+    lever presses, reward port entries, nose pokes, etc. The output text files format this data into a series of
+    colon-separated variables that are either single-line (for metadata) or multi-line (for arrays). The multi-line
+    variables keep a colon-separated index of the array every 5 elements.  For example, a single variable might look like::
+
+        Start Date: 11/09/18
+
+    while a multi-line variable might look like::
+
+        A:
+            0:      175.150      270.750      762.050      762.900     1042.600
+            5:     1567.800     1774.950     2448.450     2454.050     2552.800
+            10:     2620.550     2726.250
+
+    Different sessions are usually separated by a blank line or two.
+
+    This data is parsed by the MedPCInterface and added to the NWBFile as Events and IntervalSeries objects in the
+    behavior module.
+    """
+
+    keywords = ["behavior"]
+    display_name = "MedPC"
+    info = "Interface for handling MedPC output files."
+    associated_suffixes = ".txt"
+    default_medpc_name_to_info_dict = {}
+    default_events = []
+    default_interval_series = []
+    default_module_name = "behavior"
+    default_module_description = "Operant behavioral data from MedPC."
```

Comment:
It's certainly an interesting style to have all of these set as class attributes

I could understand it a little more if the reason was to be able to reference and control those defaults across multiple methods (such as `get_metadata` as well as `add_to_nwbfile`) but currently they only seem to be used in the `add_to_nwbfile` method, which would really imply that they are actually conversion options rather than metadata or init values

Actually, the 'medpc_name_to_info_dict' occurring in both the `__init__` signature and the instance attribute (as a default) is very confusing

I'd recommend removing these as attributes and localizing their usage within particular steps or defining a JSON 'default' in the metadata schema

Reply:
I know it's a bit clunky, but I needed a way to pass in the Events dictionary, IntervalSeries dictionary, and `medpc_name_to_info_dict` without using metadata (where they normally come from).  So, I built a backdoor with the default dicts.

I would much rather simply pass these dicts in with the metadata, but all of the checks in `test_all_conversion_checks` grab metadata from the interface `metadata = self.interface.get_metadata()`.  These dictionaries can't come from `get_metadata()` bc they are specified by the user -- not the data itself.

Reply:
That sounds like a `conversion_option` like I say above

Could you move them to that role so they are scoped and defined better?

Reply:
Also, I'm not understanding

> I would much rather simply pass these dicts in with the metadata, but all of the checks in test_all_conversion_checks grab metadata from the interface metadata = self.interface.get_metadata().

If you need to override something in the checks or define a special mixin for the new class, that's not unusual

Reply:
(which you already have anyway)